core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2496,6 +2545,367 @@ public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorCo
+                Expression errorDefault = node.getErrorDefault().get();
+                if (node.getErrorBehavior() != DEFAULT) {
+                    throw semanticException(INVALID_FUNCTION_ARGUMENT, errorDefault, "Default value specified for %s ON ERROR behavior", node.getErrorBehavior());
+                }
+                Type type = process(errorDefault, context);
+                // this would normally be done after function resolution, but we know that the default expression is always coerced to the returnedType
+                coerceType(errorDefault, type, returnedType, "Function JSON_VALUE default ON ERROR result");
+            }
+
+            // pass remaining information in the node : empty behavior, empty default, error behavior, error default
+            ImmutableList.Builder<Type> argumentTypes = ImmutableList.<Type>builder()
+                    .addAll(pathInvocationArgumentTypes)
+                    .add(TINYINT) // empty behavior: enum encoded as integer value
+                    .add(returnedType) // empty default
+                    .add(TINYINT) // error behavior: enum encoded as integer value
+                    .add(returnedType); // error default
+
+            // resolve function
+            ResolvedFunction function;
+            try {
+                function = plannerContext.getMetadata().resolveFunction(session, QualifiedName.of(JSON_VALUE_FUNCTION_NAME), fromTypes(argumentTypes.build()));
+            }
+            catch (TrinoException e) {
+                if (e.getLocation().isPresent()) {
+                    throw e;
+                }
+                throw new TrinoException(e::getErrorCode, extractLocation(node), e.getMessage(), e);
+            }
+
+            accessControl.checkCanExecuteFunction(SecurityContext.of(session), JSON_VALUE_FUNCTION_NAME);
+            resolvedFunctions.put(NodeRef.of(node), function);
+            Type type = function.getSignature().getReturnType();
+
+            return setExpressionType(node, type);
+        }
+
+        @Override
+        public Type visitJsonQuery(JsonQuery node, StackableAstVisitorContext<Context> context)
+        {
+            List<Type> pathInvocationArgumentTypes = analyzeJsonPathInvocation("JSON_QUERY", node, node.getJsonPathInvocation(), context);
+
+            // validate wrapper and quotes behavior
+            if ((node.getWrapperBehavior() == CONDITIONAL || node.getWrapperBehavior() == UNCONDITIONAL) && node.getQuotesBehavior().isPresent()) {
+                throw semanticException(INVALID_FUNCTION_ARGUMENT, node, "%s QUOTES behavior specified with WITH %s ARRAY WRAPPER behavior", node.getQuotesBehavior().get(), node.getWrapperBehavior());
+            }
+
+            // wrapper behavior, empty behavior and error behavior will be passed as arguments to function
+            // quotes behavior is handled by the corresponding output function
+            ImmutableList.Builder<Type> argumentTypes = ImmutableList.<Type>builder()
```

Comment:
Call `.build()` here directly and declare the variable as `List<Type>`. No need to have an explicit variable to hold the intermediate.